src/v/cloud_storage/async_manifest_view.cc

```diff
@@ -1602,8 +1602,8 @@ remote_manifest_path async_manifest_view::get_spillover_manifest_path(
       .base_ts = meta.base_timestamp,
       .last_ts = meta.max_timestamp,
     };
-    return generate_spillover_manifest_path(
-      get_ntp(), _stm_manifest.get_revision_id(), comp);
+    return remote_manifest_path{
```

Comment:
here the code actually uses internal path provider to return correct name instead of accepting path provider as a parameter which is different from the way partition_manifest is implemented

Reply:
That's right, I think the shift in mindset is that the manifest view and the partition manifest are not equivalent to each other. The manifest view is much more full-featured and must be able to access paths of an STM manifest and spillover manifests, unlike the partition_manifest class which is focused on tracking member fields and serialization.